src/goto-programs/show_symbol_table.cpp

```diff
@@ -68,16 +69,16 @@ void show_symbol_table_plain(
   out << '\n' << "Symbols:" << '\n' << '\n';
 
   // we want to sort alphabetically
-  std::set<std::string> symbols;
+  std::vector<irep_idt> symbols;
+  symbols.reserve(symbol_table.symbols.size());
 
   for(const auto &symbol_pair : symbol_table.symbols)
-  {
-    symbols.insert(id2string(symbol_pair.first));
-  }
+    symbols.push_back(symbol_pair.first);
+  std::sort(symbols.begin(), symbols.end(), dstring_string_cmp);
```

Comment:
I would expect this to be a lot faster -- id2string is very, very cheap (it's an array index operation); std::sort() in a vector should easily beat the std::set.

Reply:
Data, please.

(Note 1: I suppose it should be easy to take measurements before and after for some largish Java code. Measure the overall execution time and measure how much of it is spent in `show_symbol_table*`.)

(Note 2: Nominally the number of comparisons should be the same (n inserts with each O(log(n)) comparisons vs. n*log(n) during sort). Maintaining the red-black tree underlying the set is more expensive, but more id2string operations are needed on the other hand. And all this might be rather irrelevant as the execution time required to even build the symbol table might vastly dominate. Or maybe I'm completely wrong. Data will tell.)